libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -0,0 +1,175 @@
+
+    // @Group: BATUSED
+    // @Path: AP_OSD_Setting.cpp
+    AP_SUBGROUPINFO(batused, "BATUSED", 6, AP_OSD_Screen, AP_OSD_Setting),
+
+    // @Group: SATS
+    // @Path: AP_OSD_Setting.cpp
+    AP_SUBGROUPINFO(sats, "SATS", 7, AP_OSD_Screen, AP_OSD_Setting),
+
+    // @Group: FLTMODE
+    // @Path: AP_OSD_Setting.cpp
+    AP_SUBGROUPINFO(fltmode, "FLTMODE", 8, AP_OSD_Screen, AP_OSD_Setting),
+
+    // @Group: MESSAGE
+    // @Path: AP_OSD_Setting.cpp
+    AP_SUBGROUPINFO(message, "MESSAGE", 9, AP_OSD_Screen, AP_OSD_Setting),
+
+    AP_GROUPEND
+};
+
+// constructor
+AP_OSD_Screen::AP_OSD_Screen()
+{
+}
+
+//Symbols
+#define SYM_BLANK 0x20
+#define SYM_COLON 0x2D
+#define SYM_ZERO_HALF_TRAILING_DOT 192
+#define SYM_ZERO_HALF_LEADING_DOT 208
+
+#define SYM_M       177
+#define SYM_ALT_M       177
+#define SYM_BATT_FULL   0x90
+#define SYM_RSSI 0x01
+
+#define SYM_VOLT  0x06
+#define SYM_AMP   0x9A
+#define SYM_MAH   0x07
+
+#define SYM_SAT_L 0x1E
+#define SYM_SAT_R 0x1F
+
+#define AH_SYMBOL_COUNT 9
+
+void AP_OSD_Screen::draw_altitude(uint8_t x, uint8_t y)
+{
+    float alt;
+    AP_AHRS::get_singleton()->get_relative_position_D_home(alt);
+    backend->write(x, y, false, "%4.0f%c", alt, SYM_ALT_M);
+}
+
+void AP_OSD_Screen::draw_bat_volt(uint8_t x, uint8_t y)
+{
+    AP_BattMonitor &battery = AP_BattMonitor::battery();
+    uint8_t p = battery.capacity_remaining_pct();
+    p = (100 - p) / 16.6;
+    backend->write(x,y, battery.has_failsafed(), "%c%2.1fV", SYM_BATT_FULL + p, battery.voltage());
+}
+
+void AP_OSD_Screen::draw_rssi(uint8_t x, uint8_t y)
+{
+    int rssiv = AP_RSSI::get_instance()->read_receiver_rssi_uint8();
+    rssiv = (rssiv * 99) / 255;
+    backend->write(x, y, rssiv < 5, "%c%2d", SYM_RSSI, rssiv);
+}
+
+void AP_OSD_Screen::draw_current(uint8_t x, uint8_t y)
+{
+    AP_BattMonitor &battery = AP_BattMonitor::battery();
+    float amps = battery.current_amps();
+    backend->write(x, y, false, "%c%2.1f", SYM_AMP, amps);
+}
+
+void AP_OSD_Screen::draw_fltmode(uint8_t x, uint8_t y)
+{
+    backend->write(x, y, AP_Notify::instance()->get_flight_mode_str());
+}
+
+void AP_OSD_Screen::draw_sats(uint8_t x, uint8_t y)
+{
+    AP_GPS & gps = AP::gps();
+    bool has_3dfix = gps.status() <= AP_GPS::GPS_OK_FIX_2D;
+    backend->write(x, y, !has_3dfix, "%c%c%2d", SYM_SAT_L, SYM_SAT_R, gps.num_sats());
+}
+
+void AP_OSD_Screen::draw_batused(uint8_t x, uint8_t y)
+{
+    AP_BattMonitor &battery = AP_BattMonitor::battery();
+    backend->write(x,y, battery.has_failsafed(), "%c%4.0f", SYM_MAH, battery.consumed_mah());
+}
+
+void AP_OSD_Screen::draw_message(uint8_t x, uint8_t y)
+{
+    AP_Notify * notify = AP_Notify::instance();
+    if (notify) {
+        bool text_is_valid = AP_HAL::millis() - notify->get_text_updated_millis() < 20000;
+        if (text_is_valid) {
+            backend->write(x, y, notify->get_text());
+        }
+    }
+}
+
+#define DRAW_SETTING(n) if (n.enabled) draw_ ## n(n.xpos, n.ypos)
```

Comment:
I know this cleans things up a bit but it makes IDEs and searching hard to find functions. Macros are magic. In this particular case I think it's best to not use macro. Also, don't forget the { }.